server/api.go

```diff
@@ -399,7 +399,47 @@ func (a *API) oauthRedirectHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if err = a.p.store.StoreUserInWhitelist(mmUserID); err != nil {
+	tx, err := a.p.store.BeginTx()
+	if err != nil {
+		a.p.API.LogError("Unable to begin transaction", "error", err.Error())
+		http.Error(w, "Something went wrong", http.StatusInternalServerError)
+		return
+	}
+
+	if err = a.p.store.LockWhitelist(tx); err != nil {
+		a.p.API.LogError("Unable to lock whitelist", "error", err.Error())
+		http.Error(w, "Something went wrong", http.StatusInternalServerError)
+		return
+	}
+
+	defer func() {
+		if err = a.p.store.UnlockWhitelist(tx); err != nil {
+			a.p.API.LogError("Unable to unlock whitelist", "error", err.Error())
+			return
+		}
+
+		if err = a.p.store.CommitTx(tx); err != nil {
```

Comment:
why are we only committing? Shouldn't we rollback the transaction if we get any error during the transaction?

Reply:
Because there's no operation in the transaction that needs to be rolled back.

Reply:
I'm not 100% sure but I think you should still close the transaction, right?